### PR TITLE
Side menu logo and Overview header logo configuration

### PR DESCRIPTION
**Pull request by abbyhu2000**
### Description
1. Side Menu Logo

Make side menu opensearch dashboard logo to be configurable. Use config mark.defaultUrl in default mode
and mark.darkModeUrl in dark mode. 
Also replace the original door logo with default opensearch mark in default mode and dark mode opensearch mark
in dark mode. 

Screen Shot 2021-09-26 at 9 07 13 PM

 
Screen Shot 2021-09-26 at 9 07 28 PM

2. Overview Header Logo

Make overview page header logo to be configurable. Use config mark.defaultUrl in default mode
and mark.darkModeUrl in dark mode. 
Also replace the original door logo with default opensearch mark in default mode and dark mode opensearch mark
in dark mode. 

Screen Shot 2021-09-26 at 10 06 25 PM

Screen Shot 2021-09-26 at 10 06 47 PM

### Details
In default mode, a valid mark.defaultUrl is required for custom side menu logo/custom overview header logo; otherwise, default opensearch mark will be rendered;

In dark mode, a valid mark.darkModeUrl is required for custom side menu logo/custom overview header logo; otherwise, mark.defaultUrl will be rendered; if both invalid, default opensearch dark mode mark will be rendered.

Must provide a valid defaultUrl before providing darkModeUrl.
 
Signed-off-by: Abby Hu <abigailhu2000@gmail.com>

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/ui/header/collapsible_nav.tsx [hunk 1]
```diff
@@ -273,16 +316,21 @@ export function CollapsibleNav({
         {/* OpenSearchDashboards, Observability, Security, and Management sections */}
         {orderedCategories.map((categoryName) => {
           const category = categoryDictionary[categoryName]!;
+          const opensearchLinkLogo =
+            category.label === 'OpenSearch Dashboards'
```
Comment (kavilla) [inline src/core/public/chrome/ui/header/collapsible_nav.tsx:320]: can this constant be pulled out a little higher like "DEFAULT_APP_TITLE" so it's easy to change.

File: src/plugins/home/public/application/components/home.js [hunk 2]
```diff
@@ -155,6 +155,7 @@ export class Home extends Component {
           showDevToolsLink
           showManagementLink
           title={<FormattedMessage id="home.header.title" defaultMessage="Home" />}
+          branding={getServices().injectedMetadata.getBranding()}
```
Comment (kavilla) [inline src/plugins/home/public/application/components/home.js:158]: is this related to this PR?

File: src/plugins/opensearch_dashboards_overview/public/application.tsx [hunk 3]
```diff
@@ -65,6 +65,7 @@ export const renderApp = (
           newsfeed$={newsfeed$}
           solutions={solutions}
           features={features}
+          branding={core.injectedMetadata.getBranding()}
```
Comment (kavilla) [inline src/plugins/opensearch_dashboards_overview/public/application.tsx:68]: nit: can we put this into a constant defined above to match the format

File: src/core/public/chrome/ui/header/collapsible_nav.tsx (hunk 1 above)
Comment (kavilla) [reply src/core/public/chrome/ui/header/collapsible_nav.tsx:320]: actually can we copy what we are doing here src/plugins/opensearch_dashboards_overview/public/application.tsx basing it off the id. then I dont see this value changing.

File: src/plugins/home/public/application/components/home.js (hunk 2 above)
Comment (abbyhu2000) [reply src/plugins/home/public/application/components/home.js:158]: Yes. Actually the overviewHeader component is used twice in the repository. One for overview page header, another one for home page header. Since we add branding as one of the prop of overviewHeader component, we need to pass the branding here in home.js as well. 

It has no effect on home component and It will not render any logo on the home page header because of the logic within overviewHeader component.

File: src/plugins/opensearch_dashboards_overview/public/application.tsx (hunk 3 above)
Comment (abbyhu2000) [reply src/plugins/opensearch_dashboards_overview/public/application.tsx:68]: yes thx!

File: src/core/public/chrome/ui/header/collapsible_nav.tsx (hunk 1 above)
Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/collapsible_nav.tsx:320]: I put opensearchLinklogo constant there because we need the category value for each item in the map, so i think it is more convenient to put the logic inside and after we get the category value. Thx!

Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/collapsible_nav.tsx]: Yea i changed the logic so it bases off category.id.

**Review by kavilla — APPROVED** [review]

**Comment by abbyhu2000** [comment]
Screen Shot 2021-09-27 at 2 04 23 PM
